File: src/main/java/net/allape/action/MenuOpenAction.java
package net.allape.action;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.ui.MessageDialogBuilder;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.util.ui.UIUtil;
import net.allape.bus.Services;
import net.allape.window.XFTPExplorerWindowFactory;
import org.jetbrains.annotations.NotNull;

public class MenuOpenAction extends AnAction {

    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        ToolWindow toolWindow = Services.TOOL_WINDOW;
        if (toolWindow != null) {
            toolWindow.show(() -> XFTPExplorerWindowFactory.createTheToolWindowContent(e.getProject(), toolWindow));
        } else {
            MessageDialogBuilder
                    .okCancel("Initialing...", "Please wait for a while.")
                    .icon(UIUtil.getInformationIcon())
                    .yesText("Ok")
                    .ask(e.getProject());
        }
    }
}

File: src/main/kotlin/net/allape/action/MenuOpenAction.kt
package net.allape.action

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.wm.ToolWindow
import net.allape.App
import net.allape.common.Services

class MenuOpenAction : AnAction() {
    override fun actionPerformed(e: AnActionEvent) {
        e.project?.let { project ->
            val toolWindow: ToolWindow = Services.toolWindow
            toolWindow.show { App.createTheToolWindowContent(project, toolWindow) }
        }
    }
}